### PR TITLE
Add agents to openDS model

### PR DESCRIPTION
**Pull request by samleeflang**
Replace agent specific field with generic agent model.
Added the agent on all spots where we have fields to indicate a user.
I believe this approach to be in line with the GBIF UM (agent can be attached to all classes) but it does complicate the model further.

- Missing from model would be a split in first name and family/surname, should we add those fields (ABCD has them)? Could be handy when it is not always clear how the name is build up and you want to query for example WikiData with surname.
- You could say that also the institutionId/organisation/rightHolds fields can be replaced by the agent model as it also encapsulate organisations. I decided not to that as it might introduce further complexity, but you could make an argument for it.

### CODE REVIEW

**Comment by TomDijkema** [comment]
This makes things very flexible, I believe that is good for the majority, but also poses a challenge in implementation (always making sure there could be more agents present in the array).

**Review by southeo — COMMENTED** [review]
If we know what kind of agents to expect, we can enumerate them and expand the vocab as necessary.

File: data-model/shared-models/schema/agent.json [hunk 1]
```diff
@@ -44,6 +68,7 @@
     }
   },
   "required": [
+    "agentRole",
```
[the remark this answers is not in the document]
Comment (samleeflang) [reply data-model/shared-models/schema/agent.json:71]: I am hoping always. We should know if it was a collector, recorder or something else. Could make it optional just to be save but I feel like we should always be able to fill it._

File: data-model/shared-models/schema/agent.json [hunk 2]
```diff
@@ -6,35 +6,59 @@
...
       "examples": [
-        "collector"
+        "collector",
+        "preparer",
+        "identifier",
+        "recorder"
       ]
     },
     "agentType": {
       "type": "string",
```
[the remark this answers is not in the document]
Comment (samleeflang) [reply data-model/shared-models/schema/agent.json:18]: I have no idea what roles we are going to get. These I at least know that they are there but there could be others. Still we could do an enum and increase the list if we find new ones

**Comment by samleeflang** [comment]
Had some contact with the attribution group who were interested in the agent scheme. The agent object might change a bit but we will create a new PR if those changes come. That might also include an enum for the roles

File: data-model/shared-models/schema/agent.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (samleeflang) [reply data-model/shared-models/schema/agent.json:18]: Would rather keep it open for now and close it down once we have a controlled vocab

**Review by southeo — APPROVED** [review]
:detective:  (agent)

**Review by TomDijkema — APPROVED** [review]

**Pull request merged**